pkg/utils/test/testf/testf_witht.go

```diff
@@ -0,0 +1,174 @@
+package testf
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	odhClient "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/client"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+)
+
+type WithTOpts func(*WithT)
+
+func WithEventuallyTimeout(value time.Duration) WithTOpts {
+	return func(g *WithT) {
+		g.SetDefaultEventuallyTimeout(value)
+	}
+}
+
+func WithEventuallyPollingInterval(value time.Duration) WithTOpts {
+	return func(g *WithT) {
+		g.SetDefaultEventuallyPollingInterval(value)
+	}
+}
+
+func WithConsistentlyDuration(value time.Duration) WithTOpts {
+	return func(g *WithT) {
+		g.SetDefaultConsistentlyDuration(value)
+	}
+}
+
+func WithConsistentlyPollingInterval(value time.Duration) WithTOpts {
+	return func(g *WithT) {
+		g.SetDefaultConsistentlyPollingInterval(value)
+	}
+}
+
+type WithT struct {
+	ctx    context.Context
+	client *odhClient.Client
+
+	*gomega.WithT
+}
+
+func (t *WithT) Context() context.Context {
+	return t.ctx
+}
+
+func (t *WithT) Client() *odhClient.Client {
+	return t.client
+}
+
+func (t *WithT) List(
+	gvk schema.GroupVersionKind,
+	option ...client.ListOption,
+) *EventuallyValue[[]unstructured.Unstructured] {
+	return &EventuallyValue[[]unstructured.Unstructured]{
+		ctx: t.Context(),
+		g:   t.WithT,
+		f: func(ctx context.Context) ([]unstructured.Unstructured, error) {
+			items := unstructured.UnstructuredList{}
+			items.SetGroupVersionKind(gvk)
+
+			err := t.Client().List(ctx, &items, option...)
+			switch {
+			case errors.IsNotFound(err):
```

Comment:
Probably, impossible for List?

Reply:
probably yes, but I haven't found any documentation about that so I left it but I agree it is probably misleading, I'll remove it